src/routes/+layout.ts

```diff
@@ -18,41 +23,67 @@ export const load: LayoutLoad = async ({ depends, url, fetch }) => {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const nullUser: User = null as any;
 
+    // If the user is on a path that does not require authentication, we return nullUser
     const uncheckedPaths = ["/signin", "/signup", "/resetpassword"];
     const onUncheckedPath = uncheckedPaths.includes(url.pathname);
     if (onUncheckedPath) {
         return { user: nullUser };
     }
 
+    // If the user is already cached, we return the cached user
+    // If the user is not cached (undefined or null), we will fetch the user
+    const cachedUser = getCachedUser();
+    if (cachedUser !== undefined && cachedUser !== null) {
```

Comment:
```suggestion
    if (!cachedUser) {
```
Actually undefined and null should be falsy values and the user object should always be non empty and truthy?

Reply:
You're right!

Reply:
Sorry my mistake: of course it need to be 
```typescript
if (cachedUser) {
     return { user: cachedUser };
}
``` 
and not 
```typescript
if (!cachedUser) {
     return { user: cachedUser };
}
```

Reply:
Ahh. My bad. This is also the reason why the E2E tests fail. Because we now always return null or undefined and never the cached user.

Reply:
Right.